src/main/java/com/gojek/beast/models/Records.java

```diff
@@ -45,9 +45,13 @@ public Records(List<Record> records, Instant polledTime) {
     }
 
     public long getSize() {
-        return records.stream().mapToLong(record -> {
-            return record.getSize();
-        }).sum();
+        return records.stream().mapToLong(Record::getSize).sum();
+    }
+
+    public Map<Integer, Long> getRecordCountByPartition() {
```

Comment:
The changes look good, I have one question regarding memory utilization.

There are two options which we can use for getting the records count by partition:
1. call getRecordCountByPartition everytime, we iterate over records to return the value
2. call getRecordCountByPartition in the constructor and store Map<Integer, Long> as a class property. Since Records is never modified, the map also will not be modified.

Which is a better approach to take? As far as I understand with first approach, we create new map objects everytime we call the function and it will be garbage collected.
If we take the second approach, the map is stored as a class property and its created only one time.
What are your thoughts on this?

Reply:
Have implemented lazy loading for this. 👍